configuration-resources/messaging/topics.md

```diff
@@ -0,0 +1,224 @@
+
+- two agents in the same pipeline have different resource requirements, so they m,must live in separate pods 
+- some agents require to connect directly to a topic
+- an agent is marked as not "composable"
+
+### **Schema less topics**
+
+By default LangStream interprets the contents of the messages as Unicode encoded strings (UTF-8) and when an agent tries to access the message as a structure, it tries to parse the string as JSON.
+
+This means that you can write a message in the input topic as a string and read it as a JSON structure in the output topic.
+
+When an agent that expects a structure as input encounters a string that cannot
+be parsed as JSON this is handled as a regular processing failure, and you can apply the standard failure management options (like skipping unparsable messages or posting them to the deadletter queue).
+
+### **Schema management**
+
+LangStream can handle automatically the Schema associated to the topic, depending
+on the messaging broker you are using.
+
+If you are using Apache Pulsar then the Schema Registry is built-in on the Broker
+and you don't have to configure anything more.
+
+If you are using Apache Kafka then you need to configure the URL and the credentials to access a Schema Registry.
+
+This happens in the instance.yaml file in the streamingCluster configuration,
+see the [kafka](../../instance-clusters/streaming/kafka.md) cluster documentation for the actual configuration of the Schema Registry client.
+
+LangStream comes with an abstraction of the Schema management systems and allows you to write portable applications.
+
+These are the supported schema types:
+- string
+- bytes
+- avro
+
+
+```yaml
+topics:
+  - name: "schemaless-topic"
+    creation-mode: create-if-not-exists
+- name: "string-topic"
+    creation-mode: create-if-not-exists
+    schema:
+       type: string
+  - name: "bytes-topic"
+    creation-mode: create-if-not-exists
+    schema:
+       type: bytes
+  - name: "avro-topic"
+    creation-mode: create-if-not-exists
+    schema:
+        type: avro
+        schema: |
+            {
+                "type" : "record",
+                "name" : "Pojo",
+                "namespace" : "mynamespace",
+                "fields" : [ {
+                "name" : "name",
+                "type" : "string"
+                } ]
+            }
+```
+
+In case you are consuming from a topic with AVRO schema but you don't know the schema you can omit the schema definition, like this:
+
+```yaml
+topics:
+  - name: "avro-topic-auto"
+    creation-mode: create-if-not-exists
+    schema:
+        type: avro
+```
+
+The runtime gets the schema from the registry while consuming.
+
+The same applies in case you write an AVRO record: the schema will be automatically registered in the registry.
+
+
+### **Dead letter queue topics**
+
+When you mark an agent with **on-failure: deadletter**, this means that in case of error the message that was read by the input topic has to be move to a side topic
```

Comment:
```suggestion
When you mark an agent with **on-failure: deadletter**, this means that in case of error the message that was read by the input topic has to be moved to a side topic
```